blog/2024/release-updates.adoc

```diff
@@ -0,0 +1,26 @@
+:title: Backwards compatibility in Keycloak releases
+:date: 2024-08-19
+:publish: false
+:author: Stian Thorgersen
+
+There will not be a Keycloak 27.0 release, at least for some time. We are certainly not going to stop bringing new
+features and enhancements to Keycloak, but we are committed to doing so in a backwards compatible way. Hence, the
+release following Keycloak 26.0 will be 26.1 instead.
+
+This means that going forwards you will no longer need to refer to lengthy migration guides to upgrade to the latest
+and greatest release, but rather just go ahead and upgrade, and if you run into any issues open a bug and we'll make
+sure we resolve the issues quickly. Effectively, any breaking changes you discover as part of upgrading to a new minor
+version will be our problem to fix in Keycloak, rather than your problem to adopt to.
+
+In addition to bringing minor releases to Keycloak we are splitting out client libraries to be released separately.
+Client libraries such as the Keycloak admin client, JavaScript adapter, will have an initial 26.0 release, but
+after that these will be released as needed, rather than always included in the server release.
+
+For the Keycloak server we plan to do a minor release every 3 months, and a major release every 2-3 years. As before
+only the latest release will receive security and bug fixes, but upgrading to the latest release should be a lot
+simpler.
+
+For client libraries we will not have any fixed cadence of major or minor releases, but will primarily stick to
```

Comment:
For the Keycloak server, adding new REST endpoints is a backwards compatible change. I would assume we would add them in a Keycloak minor release. 

To be able to use this new endpoint with the Keycloak admin client, I expect a new version of the Keycloak admin client. To know which version of the admin client to use with that new Keycloak minor release, I'd suggest to align the admin client minor releases with the Keycloak minor releases. Otherwise people will get confused which version of the admin client to use with what version of the Keycloak server.
This procedure would lead to the admin client possibly skipping some minor releases when there is no change in the admin API in a minor release of the Keycloak server. 

If we plan not to bump the minor release version of the admin client on each Keycloak release, the release notes would need to state with what version of the admin client that release is supposed to work with. 

Would the versioning of the different clients be independent of each other? For that I assume the different clients would live in different repositories. As of today, I see one repo for all Java clients, so would they share a version number?

Reply:
As I mentioned below, I believe we should add more details to the blog post around the release scheme and most importantly support matrix of the client libraries.

Reply:
Updating